earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -37,16 +42,94 @@ ListElementTagName(const std::string &tagname)
   }
 }
 
+XMLSSize_t  initialDOMHeapAllocSize;
+XMLSSize_t  maxDOMHeapAllocSize;
+XMLSSize_t  maxDOMSubAllocationSize;
+bool doTerminate = false;
+
+class XmlParamsException : public std::exception {};
+class MinValuesNotMet : public XmlParamsException
+{
+public:
+    const char* what() const noexcept 
+    { 
+        return "Initialization Parameters must be 1024 or greater"; 
+    }
+};
+class SizeError : public XmlParamsException
+{
+public:
+    const char* what() const noexcept
+    {
+        return "Initial heap size and block allocation size must be less than the max heap size";
+    }
+};
+
+void validateXMLParameters()
+{
+    // using 1024 as the lowest setting
+    XMLSSize_t lowestBlock = 0x400;
+
+    // check to make sure they meet the minimum size
+    if (initialDOMHeapAllocSize < lowestBlock ||
+   	maxDOMHeapAllocSize < lowestBlock     ||
+	maxDOMSubAllocationSize < lowestBlock)
+    {
+	throw MinValuesNotMet();
+    }
+
+    // check to make sure that the initial size is less than the max size
+    if (maxDOMHeapAllocSize < initialDOMHeapAllocSize ||
+        maxDOMHeapAllocSize < maxDOMSubAllocationSize ||
+        initialDOMHeapAllocSize > maxDOMHeapAllocSize)
+    {
+        throw SizeError();
+    }	
+}
+
+void setDefaultValues()
+{
+   initialDOMHeapAllocSize = 0x4000;
+   maxDOMHeapAllocSize     = 0x20000;
+   maxDOMSubAllocationSize = 0x1000;
+   doTerminate             = false;
+}
 
 // This is used only in the following function
 class UsingXMLGuard
 {
   friend void InitializeXMLLibrary() throw();
 
-  UsingXMLGuard(void) throw()
-  {
-   try {
-      XMLPlatformUtils::Initialize();
+  UsingXMLGuard(void) throw() {
+    // can change this to anything, just using this for convenience
+    // format is:
+    // <initialDOMHeapAllocSize> <maxDOMHeapAllocSize> <maxDOMSubAllocationSize"
+    std::string fn("/etc/opt/google/XMLparams");
```

Comment:
Looks like this is the old method of reading the config file. Shouldn't it be updated to use khConfigFileParser? Also, I think once that's done some of the new includes can be removed (i.e., fstream).